Antidote/IncomingCallNotificationView.m

```diff
@@ -0,0 +1,148 @@
+
+@end
+@implementation IncomingCallNotificationView
+
+
+#pragma mark - LifeCycle
+- (instancetype)initWithNickname:(NSString *)nickname
+{
+    self = [super init];
+    if (! self) {
+        return nil;
+    }
+
+    _nickname = nickname;
+
+    self.backgroundColor = [UIColor grayColor];
+
+    [self createNameLabel];
+    [self createDescriptionLabel];
+    [self createDeclineCallButton];
+    [self createAcceptCallButton];
+
+    [self installConstraints];
+
+    return self;
+}
+
+#pragma mark - View setup
+- (void)createNameLabel
+{
+    self.nameLabel = [UILabel new];
+    self.nameLabel.adjustsFontSizeToFitWidth = YES;
+    self.nameLabel.textAlignment = NSTextAlignmentCenter;
+    self.nameLabel.text = self.nickname;
+    self.nameLabel.textColor = [UIColor whiteColor];
+    self.nameLabel.font = [[AppContext sharedContext].appearance fontHelveticaNeueBoldWithSize:kIncomingNameFontSize];
+
+    [self addSubview:self.nameLabel];
+
+}
+
+- (void)createDescriptionLabel
+{
+    self.descriptionLabel = [UILabel new];
+    self.descriptionLabel.text = NSLocalizedString(@"is calling", @"Calls");
+    self.descriptionLabel.textAlignment = NSTextAlignmentCenter;
+    self.descriptionLabel.textColor = [UIColor whiteColor];
+    self.descriptionLabel.font = [[AppContext sharedContext].appearance fontHelveticaNeueBoldWithSize:kIncomingIsCallingFontSize];
+
+    [self addSubview:self.descriptionLabel];
+}
+
+- (void)createDeclineCallButton
+{
+    self.declineButton = [UIButton buttonWithType:UIButtonTypeCustom];
+    UIImage *declineCallImage = [UIImage imageNamed:@"call-decline"];
+    [self.declineButton setImage:declineCallImage forState:UIControlStateNormal];
+    self.declineButton.tintColor = [UIColor whiteColor];
+    self.declineButton.layer.borderColor = [UIColor blackColor].CGColor;
+    self.declineButton.layer.borderWidth = kButtonBorderWidth;
+    self.declineButton.backgroundColor = [[AppContext sharedContext].appearance callRedColor];
+    self.declineButton.layer.cornerRadius = kButtonSize / 2.0;
+    [self.declineButton addTarget:self action:@selector(tappedDeclineButton) forControlEvents:UIControlEventTouchUpInside];
+    [self addSubview:self.declineButton];
+}
+
+- (void)createAcceptCallButton
+{
+    self.acceptButton = [UIButton buttonWithType:UIButtonTypeCustom];
+    UIImage *acceptCallimage = [UIImage imageNamed:@"call-phone"];
+    [self.acceptButton setImage:acceptCallimage forState:UIControlStateNormal];
+    self.acceptButton.tintColor = [UIColor whiteColor];
+    self.acceptButton.layer.borderColor = [UIColor blackColor].CGColor;
+    self.acceptButton.layer.borderWidth = kButtonBorderWidth;
+    self.acceptButton.backgroundColor = [[AppContext sharedContext].appearance callGreenColor];
+    self.acceptButton.layer.cornerRadius = kButtonSize / 2.0;
+    [self.acceptButton addTarget:self action:@selector(tappedAcceptButton) forControlEvents:UIControlEventTouchUpInside];
```

Comment:
You can move creation code for decline and accept button to single method and call it from here:
```
- (UIButton *)createButtonWithImageName:(NSString *)imageName backgroundColor:(UIColor *)backgroundColor;
```